### PR TITLE
MODDATAIMP-119 Added support for incoming json files containing MARC records

### PR DESCRIPTION
**Pull request by KaterynaSenchenko**

### CODE REVIEW

File: src/main/java/org/folio/service/processing/reader/MarcJsonReader.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+      LOGGER.error("Error reading next record", e);
+      throw new RecordsReaderException(e);
+    }
+    return recordsBuffer.getRecords();
+  }
+
+  @Override
+  public boolean hasNext() {
+    try {
+      boolean value = reader.hasNext();
```
Comment (Igor-Gorchakov) [inline src/main/java/org/folio/service/processing/reader/MarcJsonReader.java]: boolean hasNext

File: NEWS.md [hunk 2]
```diff
@@ -1,5 +1,11 @@
-## 2019-03-21 v1.3.0-SNAPSHOT
+## 2019-05-02 v1.3.0-SNAPSHOT
```
Comment (Igor-Gorchakov) [inline NEWS.md]: Do we need to change a date here ?

File: src/main/java/org/folio/service/processing/reader/MarcJsonReader.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+import com.google.gson.stream.JsonToken;
+import io.vertx.core.logging.Logger;
+import io.vertx.core.logging.LoggerFactory;
+import org.apache.commons.io.FileUtils;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.List;
+
```
Comment (Igor-Gorchakov) [inline src/main/java/org/folio/service/processing/reader/MarcJsonReader.java]: Please leave a simple java doc

File: src/main/java/org/folio/service/processing/reader/RecordsReaderException.java [hunk 4]
```diff
@@ -0,0 +1,9 @@
+package org.folio.service.processing.reader;
+
```
Comment (Igor-Gorchakov) [inline src/main/java/org/folio/service/processing/reader/RecordsReaderException.java]: Please leave a simple javadoc

File: src/main/java/org/folio/service/processing/reader/SourceReaderBuilder.java [hunk 5]
```diff
@@ -22,8 +22,12 @@ public static SourceReader build(File file, JobProfileInfo jobProfile) {
     if (jobProfile != null
       && jobProfile.getDataType() != null
       && jobProfile.getDataType().equals(JobProfileInfo.DataType.MARC)) {
+      String name = file.getName();
+      if (JSON_EXTENSION.equals(name.substring(name.indexOf('.')))) {
+        return new MarcJsonReader(file, CHUNK_SIZE);
+      }
       return new MarcSourceReader(file, CHUNK_SIZE);
```
Comment (Igor-Gorchakov) [inline src/main/java/org/folio/service/processing/reader/SourceReaderBuilder.java]: Maybe better to rename MarcSourceReader to MarcRawReader ?

File: src/main/java/org/folio/service/processing/reader/SourceReaderBuilder.java [hunk 6]
```diff
@@ -22,8 +22,12 @@ public static SourceReader build(File file, JobProfileInfo jobProfile) {
     if (jobProfile != null
       && jobProfile.getDataType() != null
       && jobProfile.getDataType().equals(JobProfileInfo.DataType.MARC)) {
+      String name = file.getName();
+      if (JSON_EXTENSION.equals(name.substring(name.indexOf('.')))) {
```
Comment (Igor-Gorchakov) [inline src/main/java/org/folio/service/processing/reader/SourceReaderBuilder.java]: name.lastIndexOf('.')

File: src/main/java/org/folio/service/processing/reader/MarcJsonReader.java [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+  private static final Logger LOGGER = LoggerFactory.getLogger(MarcJsonReader.class);
+  private JsonReader reader;
+  private int chunkSize;
+
+  public MarcJsonReader(File file, int chunkSize) {
+    this.chunkSize = chunkSize;
+    try {
+      this.reader = new JsonReader(new InputStreamReader(FileUtils.openInputStream(file)));
+    } catch (IOException e) {
+      LOGGER.error("Cannot initialize reader", e);
```
Comment (Igor-Gorchakov) [inline src/main/java/org/folio/service/processing/reader/MarcJsonReader.java]: Please log some information to understand for what file the exception is thrown - file name or file path

File: NEWS.md [hunk 8]
```diff
@@ -1,5 +1,11 @@
...
 * Added Spring DI support
+* Added dependency on users interface
```
Comment (Igor-Gorchakov) [inline NEWS.md]: Do we have dependency on mod-users in module-descriptor ('requires' section)?

File: src/main/java/org/folio/service/processing/reader/MarcJsonReader.java [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+  }
+
+  @Override
+  public List<String> next() {
+    RecordsBuffer recordsBuffer = new RecordsBuffer(this.chunkSize);
+    try {
+      Gson gson = new GsonBuilder().create();
+      if (reader.peek().equals(JsonToken.BEGIN_ARRAY)) {
+        reader.beginArray();
+      }
```
Comment (Igor-Gorchakov) [inline src/main/java/org/folio/service/processing/reader/MarcJsonReader.java]: I expected to see an 'else' section here (with preparation reader to work with single json object)

**Comment by Igor-Gorchakov** [comment]
Looks cool

File: NEWS.md (hunk 2 above)
Comment (KaterynaSenchenko) [reply NEWS.md]: it doesn't really matter, we will update the date anyway to reflect the release date

File: NEWS.md (hunk 8 above)
Comment (KaterynaSenchenko) [reply NEWS.md]: yep) Oleksii added the dependency, but did not update the News

File: src/main/java/org/folio/service/processing/reader/MarcJsonReader.java (hunk 9 above)
Comment (KaterynaSenchenko) [reply src/main/java/org/folio/service/processing/reader/MarcJsonReader.java]: no need, JsonReader recognizes single json object by itself

**Pull request merged**